cmd/kots/cli/plugin.go

```diff
@@ -0,0 +1,201 @@
+package cli
+
+import (
+	"fmt"
+	"io/ioutil"
+	"net/url"
+	"os"
+
+	"github.com/pkg/errors"
+	"github.com/replicatedhq/kots/pkg/kotsutil"
+	"github.com/replicatedhq/kots/pkg/logger"
+	"github.com/replicatedhq/kots/pkg/preflight"
+	"github.com/replicatedhq/kots/pkg/template"
+	analyzerunner "github.com/replicatedhq/troubleshoot/pkg/analyze"
+	"github.com/replicatedhq/troubleshoot/pkg/oci"
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+)
+
+func PluginCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:           "plugin",
+		Short:         "Run commands for helm plugin",
+		Long:          `Run commands for helm plugin, this serves as an entry point for subcommands such as preflight`,
+		SilenceUsage:  false,
+		SilenceErrors: false,
+		PreRun: func(cmd *cobra.Command, args []string) {
+			viper.BindPFlags(cmd.Flags())
+		},
+		RunE: func(cmd *cobra.Command, args []string) error {
+			if len(args) != 1 {
+				cmd.Help()
+				os.Exit(1)
+			}
+			log := logger.NewCLILogger(cmd.OutOrStdout())
+
+			if args[0] == "preflight" {
+				preflightFile, err := cmd.Flags().GetString("preflight")
+				if err != nil {
+					log.Errorf("failed to get preflight flag: %v\n", err)
+					return err
+				}
+
+				configFile, err := cmd.Flags().GetString("config")
+				if err != nil {
+					log.Errorf("failed to get config flag: %v\n", err)
+					return err
+				}
+
+				err = renderAndRunPreflight(preflightFile, configFile)
+				if err != nil {
+					log.Errorf("failed to render and run preflight %s: %v\n", preflightFile, err)
+					return err
+				}
+			} else if args[0] == "config" {
+				// TODO: logic for initial install for pulling config yaml
+			}
+			return nil
+		},
+	}
+
+	cmd.Flags().String("preflight", "", "the filename or url of the Preflight spec")
```

Comment:
These subcommands,  not flags.

Reply:
they were flags for preflight spec and config spec..definitely will pick better naming since we are going to have preflight and config subcommands